manager/integration/tests/test_backing_image.py

```diff
@@ -694,3 +704,199 @@ def test_volume_wait_for_backing_image_condition(client): # NOQA
     volume_endpoint = get_volume_endpoint(volume2)
     vol2_cksum = get_device_checksum(volume_endpoint)
     assert vol1_cksum == vol2_cksum
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_min_number_of_replicas(client): # NOQA
+    """
+    Test the backing image minNumberOfCopies
+
+    Given
+    - Create a BackingImage
+    - Update the minNumberOfCopies to 2
+
+    When
+    - The BackingImage is prepared and transferred to manager
+
+    Then
+    - There will be two ready file of the BackingIamge
+
+    Given
+    - Set the setting backing-image-cleanup-wait-interval to 1 minute
+
+    When
+    - Update the minNumberOfCopies to 1
+    - Wait for 1 minute
+
+    Then
+    - There will be one ready BackingImage file left.
+    """
+    bi_url = BACKING_IMAGE_RAW_URL
+    create_backing_image_with_matching_url(
+        client, BACKING_IMAGE_NAME, bi_url)
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=2)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 2, "ready")
+
+    update_setting(client, "backing-image-cleanup-wait-interval", "1")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=1)
+
+    time.sleep(60)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_selector_setting(client, volume_name): # NOQA
+    """
+    - Set node1 with nodeTag = [node1], diskTag = [node1]
+    - Set node2 with nodeTag = [node2], diskTag = [node2]
+    - Create a BackingImage with
+        - minNumberOfCopies = 2
+        - nodeSelector = [node1]
+        - diskSelector = [node1]
+    - After creation, the first BackingImage copy will be on node1, disk1
+    - Wait for a while, the second one will never show up
```

Comment:
What will happen for the case
```
- Set node1 with nodeTag = [node1], disk1 diskTag = [ssd], disk2 diskTag = [ssd] 
- Set node2 with nodeTag = [node2], diskTag = [ssd]

    - Create a BackingImage with
        - minNumberOfCopies = 2
        - nodeSelector = [node1]
        - diskSelector = [ssd]
```
Can we allow the two copies on the same node's different disks?

Reply:
yes we can
backing image manager is disk level
and just like replicas, it allows to create file on the different disks of the same node

Reply:
after all, if two replicas on disk1 and disk2 of the same node node1
we still need to prepare two backing image file on disk1 and disk2